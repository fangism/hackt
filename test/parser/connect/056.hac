// "056.hac" should fail: type mismatch

defproc portaproc(bool b, c[4]; int i, j[2]);

bool a;
bool x[4];
int q;
int r[2];

portaproc p, s;

p(x[2], x, a, r);		// type mismatch
s( ,  ,  ,  );

