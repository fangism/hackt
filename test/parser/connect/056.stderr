Meta-class type does not match!  got: int-inst-ref and: bool-inst-ref
ERROR: actual instance reference 3 of port connection (a) doesn't match the formal (portaproc::i) type/size.
	got: bool<>, expected: int<32>
At least one error in port connection of (portaproc<> p)
HALT: at least one error in port connection list.  [12:2..16]
